### PR TITLE
Update concept article and AccessLevel numbers

### PR DESCRIPTION
**Pull request by Cynerd**

### CODE REVIEW

File: src/rpcmessage.md [hunk 1]
```diff
@@ -47,6 +47,24 @@ The `ShvPath` is used to select exact node of method in the SHV tree.
 `Access` is part of access control. It is assigned to request by broker according to user rights.
 Multiple grants can be specified and separated by comma. 
 
+`AccessLevel` is new way to specify access level. It is numerical with
+predefined range (0-63) and brokers on the way can lower this number to even
+further limit access. Broker can't increase this number. `Access` should be used
+to get level if this field is not present and if even that is not present then
+Admin access should be considered.
```
Comment (fvacek) [inline src/rpcmessage.md:54]: > `Access` should be used to get level if this field is not present and if even that is not present then
Admin access should be considered.

Does it mean that if both fields `Access` and `AccessLevel` are not present, then `Access == su` should be considered?

File: src/rpcmessage.md [hunk 2]
```diff
@@ -47,6 +47,24 @@ The `ShvPath` is used to select exact node of method in the SHV tree.
...
+
+| Name          | Numerical representation | `Access` representation |
+|---------------|--------------------------|-------------------------|
```
Comment (fvacek) [inline src/rpcmessage.md:57]: 0 - no access

File: src/rpcmethods/file.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+This method provides information about this file. It is required for file nodes.
+
+| Parameter | Result |
+|-----------|--------|
+| Null      | i{...} |
+
+The result is IMap with these fields:
+
+| Key | Type             | Description                                                                            |
```
Comment (fvacek) [inline src/rpcmethods/file.md:29]: Key names would be fine
```
| Key | Name       | Type             | Description                                                                            |
|-----|------------|------------------|----------------------------------------------------------------------------------------|
| 0   | Type       | Int              | Type of the file (at the moment only regular is supported and thus must always be `0`) |
| 1   | Size       | Int              | Size of the file in bytes                                                              |
| 2   | PageSize   | Int              | Page size (ideal size and thus alignment for this file efficient access)               |
| 3   | AccessTime | DateTime \| Null | Optional time of latest data access                                                    |
| 4   | ModTime    | DateTime \| Null | Optional time of latest data modification                                              |
```

File: src/rpcmethods/file.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+| Parameter | Result |
+|-----------|--------|
+| Null      | Int    |
+
+```
+=> <id:42, method:"size", path:"test/file">i{}
+<= <id:42>i{2:3674}
+```
+
+## `*:crc`
```
Comment (fvacek) [inline src/rpcmethods/file.md:62]: We need also SHA1, so we can either rename this function to `checksum` with one more param or introduce new function `sha1`.

File: src/shvrpcconcepts.md [hunk 5]
```diff
@@ -33,110 +47,80 @@ in parallel in such case. Clients are only connected to the broker.
...
+subscription and if message matches some, then it is propagated to that client.
+The SHV path of the message is prefixed with mount point of the client the
+signal was received from. This way other clients see signals as being delivered
+from the correct place in the SHV tree. Signals received from not-mounted
+clients are simply dropped.
+
+The API client's have available from Broker is documented in [Broker
+API](./rpcmethods/broker.md).
+
+Brokers must maintain their own subscriptions in mounted clients that are broker
```
Comment (fvacek) [inline src/shvrpcconcepts.md:95]: Doesn't `rejectNotSubscribed` cover this?

File: src/shvrpcconcepts.md [hunk 6]
```diff
@@ -33,110 +47,80 @@ in parallel in such case. Clients are only connected to the broker.
...
+request to the broker, and it assigns to the message some access level based on
+users rules. The message is then delivered to the mounted client that checks
+this level and either performs the method or raises error based on it. Brokers
+can choose to handle request with error if not even minimal browse access is
+given to the client.
+
+The predefined access levels are the following:
+
+| Name          | Description                                                                                                                                                  |
+|---------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (fvacek) [inline src/shvrpcconcepts.md:114]: I think that we should keep `su`, `dev`, ..., `bws` strings introduced until `access` tag is not dropped.

File: src/rpcmessage.md (hunk 1 above)
Comment (Cynerd) [reply src/rpcmessage.md:54]: Yes, but that is only a case when there is no broker in between, so it is a safe and correct default. Peers without access rights should have full access. If you want to limit peer then such peer must perform authentication and thus you must include at least that part of the broker functionality in your application. Also the important word here is "should" and not "must".  If there is a reason for limiting access level outside of the SHV RPC message, then it can be done.

File: src/rpcmessage.md (hunk 2 above)
Comment (Cynerd) [reply src/rpcmessage.md:57]: Is that really needed to be defined? I kept it empty for implementations to fill it in if they have need for that, but message with this level can't exists because broker should send error about missing node instead.

The only use case would be when we would have some sort of access level query method in the broker defined, but we do not have that and thus it should not be in the standard. Standard should contain only definition that must be there (used by standard or reserved) and nothing else in my eyes.

File: src/rpcmethods/file.md (hunk 4 above)
Comment (Cynerd) [reply src/rpcmethods/file.md:62]: Why? In what way it provides a better functionality over crc32? It has lower probability of conflict but by combining few more crc32 calculations you can get the same (multiple crc32 values must match such s crc32 of the whole file, crc32 of its first half and crc32 of its second half).

`sha1` is not enough for security anyway.

I am not strictly against that. We can add it with condition that it can additionally optionally implemented if need be but the only ensured way us the crc32.

File: src/shvrpcconcepts.md (hunk 5 above)
Comment (Cynerd) [reply src/shvrpcconcepts.md:95]: That is not about removing subscriptions strictly speaking. This is about doing steps so subscribed notifications are also propagated from subbrokers. That includes subscriptions to the subbroker not only removal of old subscriptions.

And I removed the previous article that described `rejectNotSubscribed` because that usage is invalid anyway once you consider glob subscribes, but that is outside of this discussion.

File: src/shvrpcconcepts.md (hunk 6 above)
Comment (Cynerd) [reply src/shvrpcconcepts.md:114]: This article is about concepts. It should not really contain format definitions. It is for first time users and should not serve as a data definition.

TLDR: it wasn't removed, it was moved to better location that is rpcmessage article.

File: src/rpcmessage.md (hunk 1 above)
Comment (fvacek) [reply src/rpcmessage.md:54]: Understand, can be this discrepancy between broker/direct connection mentioned in paragraph above pls?

File: src/rpcmessage.md (hunk 2 above)
Comment (fvacek) [reply src/rpcmessage.md:57]: > The only use case would be when we would have some sort of access level query method in the broker defined

I'm afraid that this should be added to standard, because we are using it currently in FL.

File: src/rpcmessage.md (hunk 1 above)
Comment (Cynerd) [reply src/rpcmessage.md:54]: It is not a disrepancy. It is just how that is covered. Please read the paragraph in the documentation again. It clearly states that brokers decreases this. It is clear that you must start with highest level to allow this to work.

In only addressed the direct connection from point of security here and I do not think that security analysis should be in this article (although there might be one somewhere else, but I do not think that you would like the result of it).

File: src/rpcmessage.md (hunk 2 above)
Comment (Cynerd) [reply src/rpcmessage.md:57]: We were already talking about that. We can add it but if so let's add it outside of this PR that already contains a lot of changes.

I have nothing against adding it but we must standardize method that uses it. That is why I don't want to add it right here and now.

File: src/rpcmessage.md (hunk 1 above)
Comment (fvacek) [reply src/rpcmessage.md:54]: Well again
> Access should be used to get level if this field is not present and if even that is not present then
Admin access should be considered.

We are talking about case when neither `Access` nor `AccessLevel` is specified. According to above, `then Admin access should be considered`. What will be implicitly filled into `Access` and `AccessLevel` fields in that case? This is something what is not clearly understandable from the sentence above.

Comment (fvacek) [reply src/rpcmessage.md:54]: What about


`AccessLevel` is new way to specify access level. It is numerical with
predefined range (0-63) and brokers on the way can lower this number to further limit access. Broker can't increase this number. `Access` should be used
to get level if this field is not present. If neither `AccessLevel` nor `Access` field is present then
Admin `AccessLevel` should be considered as the base limit.

File: src/rpcmethods/file.md (hunk 4 above)
Comment (Cynerd) [reply src/rpcmethods/file.md:63]: `sha1` added.
